src/lib/src/address.cpp

```diff
@@ -67,6 +67,46 @@ int roc_address_init(roc_address* address, roc_family family, const char* ip, in
     return -1;
 }
 
+int roc_address_set_multicast_interface(roc_address* address,
+                                        roc_family family,
+                                        const char* iface) {
+    if (!address) {
+        return -1;
+    }
+
+    if (!iface) {
+        return -1;
+    }
+
+    packet::Address& pa = get_address(address);
+
+    if (!pa.multicast()) {
+        return -1;
+    }
+
+    if (family == ROC_AF_AUTO || family == ROC_AF_IPv4) {
+        if (pa.set_multicast_iface_v4(iface)) {
+            if (pa.version() == 6) {
+                return -1;
```

Comment:
This leaves the address in inconsistent state. If we can't proceed, it's better to keep the address unmodified.

Reply:
I did it because we have `ROC_AF_AUTO` and we actually don't know the exact family. We try to set multicast for a ipv4 address and if it fails we will try to set it to the ipv6 address.

From another side, we can remove `family` parameter from the argument list and say the we will set the multicast based on the network address (initialized with roc_address_init). This will simplify the API and the code.

Reply:
Discussed to remove roc_address_set_multicast_interface and merge this functionality into roc_address_init.